Feature local build

lpugin opened the pull request:
Enable the GH Action to deploy the Schema and the Guidelines

bwbohl committed:
add TEI Stylesheets released branch as submodule

bwbohl committed:
add TEI Stylesheets released branch as submodule

bwbohl committed:
add TEI Stylesheets released branch as submodule

bwbohl committed:
add TEI Stylesheets released branch as submodule

bwbohl committed:
Set submodule TEI Stylesheets to commit used in travis

https://github.com/TEIC/Stylesheets/commit/c5d26c36f5beb01688c75591e8cc1c9c7e34b039

bwbohl committed:
Set submodule TEI Stylesheets to commit used in travis

https://github.com/TEIC/Stylesheets/commit/c5d26c36f5beb01688c75591e8cc1c9c7e34b039

bwbohl committed:
Set submodule TEI Stylesheets to commit used in travis

https://github.com/TEIC/Stylesheets/commit/c5d26c36f5beb01688c75591e8cc1c9c7e34b039

bwbohl committed:
Set submodule TEI Stylesheets to commit used in travis

https://github.com/TEIC/Stylesheets/commit/c5d26c36f5beb01688c75591e8cc1c9c7e34b039

bwbohl committed:
Update .gitignore

add lib directory to ignores

bwbohl committed:
Update .gitignore

add lib directory to ignores

bwbohl committed:
Update .gitignore

add lib directory to ignores

bwbohl committed:
Create build.xml

bwbohl committed:
Create build.xml

bwbohl committed:
Create build.xml

bwbohl committed:
Create build.xml

bwbohl committed:
add target to build all customizations

bwbohl committed:
add target to build all customizations

bwbohl committed:
add target to build all customizations

bwbohl committed:
Merge branch 'feature_local-build' of https://github.com/music-encoding/music-encoding into feature_local-build

bwbohl committed:
add xerces download

bwbohl committed:
add xerces download

bwbohl committed:
add task for transforming guidelines

bwbohl committed:
add task for transforming guidelines

bwbohl committed:
add xerces download to init targe

bwbohl committed:
add xerces download to init targe

bwbohl committed:
fix documentation

bwbohl committed:
fix documentation

bwbohl committed:
some formatting of build.xml

bwbohl committed:
some formatting of build.xml

bwbohl committed:
Update build.xml

add reset target to delete lib and temp folders

bwbohl committed:
Update build.xml

add reset target to delete lib and temp folders

bwbohl committed:
add canonicalize-source ant task

bwbohl committed:
add canonicalize-source ant task

bwbohl committed:
add target description for canonicalize-source

bwbohl committed:
add target description for canonicalize-source

bwbohl committed:
Update build.xml

add some commandline info on build-rng

bwbohl committed:
Update build.xml

add some commandline info on build-rng

bwbohl committed:
Update build.xml

remove property selectedSchema as this resulted in empty generated schema

bwbohl committed:
Update build.xml

remove property selectedSchema as this resulted in empty generated schema

bwbohl committed:
Update build.xml

fix path in build-compiled-odd

bwbohl committed:
Update build.xml

fix path in build-compiled-odd

bwbohl committed:
Update copy.xsl

add some formatting for teiTEI and xml-model processing-instructions

bwbohl committed:
Update copy.xsl

add some formatting for teiTEI and xml-model processing-instructions

bwbohl committed:
Update build.xml

make build-rng depend on canonicalize-source

bwbohl committed:
Update build.xml

make build-rng depend on canonicalize-source

bwbohl committed:
Update build.xml

remove erroneuos line accidentally commited

bwbohl committed:
Update build.xml

remove erroneuos line accidentally commited

bwbohl committed:
Update build.xml

fix build-compiled-odd

bwbohl committed:
Update build.xml

fix build-compiled-odd

bwbohl committed:
Update build.xml

removed unused arg

bwbohl committed:
Update build.xml

removed unused arg

bwbohl committed:
Update build.xml

add dist target

bwbohl committed:
Update build.xml

add dist target

bwbohl committed:
Update build.xml

make dist trget default

bwbohl committed:
Update build.xml

make dist trget default

bwbohl committed:
Update build.xml

make dist target depend on init target

bwbohl committed:
Update build.xml

make dist target depend on init target

bwbohl committed:
Merge branch 'feature_local-build' of https://github.com/music-encoding/music-encoding into feature_local-build

bwbohl committed:
delete dist folder on ant reset

bwbohl committed:
update ant documentation

bwbohl committed:
add build target for guidelines web

bwbohl committed:
comment-out unused target

bwbohl committed:
add TODOs to build.xml

bwbohl committed:
Update .gitignore

add dist folder to ignores
remove build folder from ignores

bwbohl committed:
Update build.xml

update project properties

bwbohl committed:
Update build.xml

make targets use new parameters

bwbohl committed:
Update build.xml

avoid reexecution of init targets

bwbohl committed:
Update build.xml

fix order of parameters and remove trailing slash in dir.dist.schemata

bwbohl committed:
Create ant.yml

bwbohl committed:
Merge pull request #736 from music-encoding/bwbohl-ghaction-ant

Create ant.yml

bwbohl committed:
Update ant.yml

bwbohl committed:
Update ant.yml

checkout submodules

lpugin committed:
Adjust script to upload schema (WIP)

lpugin committed:
Fix indentation

lpugin committed:
Change branch name

lpugin committed:
Adding -r option to cp

lpugin committed:
Add guidelines

lpugin committed:
Adding missing directory value

lpugin committed:
Fixing ant script

lpugin committed:
copy ./dev/web (and not ./dev)

lpugin committed:
mv ./web to ./dev before pushing

bwbohl committed:
fix some indentation

bwbohl committed:
Update build.xml

replace copydir wirth copy task

lpugin committed:
Fix typo

lpugin committed:
Rename workflow file and change branches deploy branches to master

lpugin committed:
Remove unused travis files

lpugin committed:
Merge branch 'develop' into feature_local-build

kepper commented on .github/workflows/deploy.yml:
```diff
@@ -0,0 +1,131 @@
+env:
+  # schema
+  SCHEMA_REPO: ${{ github.repository_owner }}/schema
+  SCHEMA_BRANCH: feature_local-build
```
Is this the correct branch?

lpugin replied:
Well spotted. I changed it.

kepper reviewed (commented):
Not sure about the branch on the `schema` repo. And I haven't tested this locally (need to familiarize myself with this first…)

lpugin committed:
Fix branch for schema

lpugin committed:
Change guidelines dir

kepper commented on build.xml:
```diff
@@ -0,0 +1,175 @@
+            <arg value="-xsl:${basedir}/utils/guidelines_xslt/odd2html.xsl" />
+            <arg value="-o:${dir.dist.guidelines}" />
+            <arg value="-xi:on"/>
+            <arg value="output.folder=${dir.dist.guidelines}/"/>
+            <arg value="git.sha=git.sha"/>
+            <arg value="branch=develop"/>
```
I assume those are the parameters passed into the XSLT, right? If that is the case, `output.folder` is correct, but `git.sha` is expected to be `hash`, and `branch` is probably supposed to be `version`, even though the value given here doesn't really fit (I'd expect something like `4.1.0-dev` or similar). @bwbohl, could you verify? We can still decide where to change things, but I'd like to understand first…

lpugin replied:
I assume we need to pass it to the ant script. I can add this later if you don't mind.

bwbohl replied:
Forgot about the sha in the first place, thx for adding. I thought $version is supposed to be dev, v4, v3 etc like in the guidelines XSLT and in ANT is used to name the dist subfolder

bwbohl replied:
ok fixed git sha in ant and XSLT relation, still have to submit it to the ant call from the GHAction, though

bwbohl replied:
I wonder about the XSLT's $version parameter and if we actually shouldn't stick to the semantic versioning concept of adding the commit sha behind the last stable version, e.g.:
`4.0.1+e429146`

or even add the develop branch-name as "pre-release info": 
`4.0.1-feature_local-build+e429146`
Of course one could argue that that the GitHub SHA already serves as a pointer to the correct branch, nevertheless this would make the version more human-readable and could come in handy with pull requests.

`git describe --tags` also allows to output the number of commits since a tag, e.g.: 
`v4.0.1-269-ge429146`
Which would give a hint on how different the version is from the last release.

or `git describe --tags --dirty` even giving dirty state if any file has been modified locally:
`v4.0.1-269-ge429146-dirty`

Nevertheless this does not fully comply to http://semver.org and should be turned into something like:
`v4.0.1-269+g.e429146`

resp.
`v4.0.1-269+g.e429146-dirty`

@music-encoding/core-contributors : any opinions?

lpugin replied:
Where is this version number then used? My understanding was that we wanted to have somewhere a way to go back to the commit of the source from which the guidelines where built from. We definitely need this, but I am fine to have more if you think we need to as long as we have that and that the output folders unchanged (v4, v5, dev).

(as I said, we can always improve this later but I think for now it would be good if we can merge this it in order to have it run so the users have back what we had before)

kepper replied:
The `version` parameter is only partially necessary. When not provided, I will extract the version from the `@meiversion` attribute. The problem there us we need to make sure this is maintained properly. The version number found there is automatically processed into the `/v4/`-style folder names. We may need to agree on some strict naming conventions for the dev version, because there it's not equally easy to get that name. 

The `hash` parameter (that @bwbohl passes in as `git.sha` – I'm fine with either way) is needed to add a link to the Git repo. Here, we just need to take a decision about the name of that parameter. Benni, your choice…

bwbohl replied:
concerning the branch arg, we don't really need this at the moment, as the XSLT isn't using it; and version is populated by the XSLT from the `@meiversion`defaultVal

kepper reviewed (commented):
Hi Benni, could you have a look at this?

bwbohl committed:
Merge pull request #738 from music-encoding/fix-copyTask

Fix copy task

bwbohl committed:
fix copying images directory to guidelines dist

bwbohl committed:
add github.sha as ant parameter

bwbohl committed:
add clean target to ant

bwbohl committed:
Update build.xml

add target for retrieving local git sha and dirty status

bwbohl committed:
Update build.xml

conditionally fill $hash XSLT argument with externally supplied github.sha or if not supplied with local git sha

bwbohl committed:
Update odd2html.xsl

output hash in initial transfomr message

bwbohl committed:
Update deploy.yml

supply github.sha in ant-call

bwbohl committed:
Update build.xml

calculate github.ref-name and github.ref-type properties from github.ref

bwbohl committed:
Update build.xml

comment unused XSLT arg

bwbohl committed:
Update build.xml

remove clean dependency on canonicalize-source target
